krm-functions/dnn-fn/lib.go

```diff
@@ -0,0 +1,100 @@
+/*
+ Copyright 2023 The Nephio Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	infrav1alpha1 "github.com/nephio-project/api/infra/v1alpha1"
+	nephioreqv1alpha1 "github.com/nephio-project/api/nf_requirements/v1alpha1"
+	ipamv1alpha1 "github.com/nokia/k8s-ipam/apis/alloc/ipam/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var theScheme *runtime.Scheme = runtime.NewScheme()
+
+func init() {
```

Comment:
do you still need the schema stuff?

Reply:
Here is an example how I did it in the interface-fn. You basically know the types based on GVK as it is 1:1 mapping

ipallocs := objs.Where(fn.IsGroupVersionKind(ipamv1alpha1.IPAllocationGroupVersionKind))
	for _, ipalloc := range ipallocs {
		if ipalloc.GetName() == forObj.GetName() {
			alloc, err := kubeobject.NewFromKubeObject[ipamv1alpha1.IPAllocation](ipalloc)
			if err != nil {
				return nil, err
			}
			allocGoStruct, err := alloc.GetGoStruct()
			if err != nil {
				return nil, err
			}
			itfce.Status.IPAllocationStatus = &allocGoStruct.Status
		}
	}

Reply:
Btw with latest lib, the above doesn't work instead it has to be: 
`objs.Where(fn.IsGroupVersionKind(schema.GroupVersionKind(ipamv1alpha1.IPAllocationGroupVersionKind)))`